test/Microsoft.AspNetCore.Mvc.ApiExplorer.Test/DefaultApiDescriptionProviderTest.cs

```diff
@@ -1856,5 +1883,10 @@ private interface ITestService
         {
 
         }
+
+        private class FromFormFileAttribute : Attribute, IBindingSourceMetadata
+        {
+            public BindingSource BindingSource => BindingSource.FormFile;
+        }
```

Comment:
Is this necessary any more?

Reply:
Sort of. We use reflection to infer sources for parameters. It's cleaner than special casing ActionDescrpition generation for this one test.